core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -399,6 +399,8 @@ class SparkContext(config: SparkConf) extends Logging {
     // This should be set as early as possible.
     SparkContext.fillMissingMagicCommitterConfsIfNeeded(_conf)
 
+    JavaModuleUtils.supplementJava17ModuleOptsIfNeeded(_conf)
```

Comment:
Do you have any better suggestions about  supplement `--add-opens` configurations to `spark.driver.extraJavaOptions` and `spark.executor.extraJavaOptions` for test cases using `local` and l`ocal-cluster` modes.

@dongjoon-hyun @wangyum

Reply:
I'm investigating whether can remove this call

Reply:
This call may not be necessary. I'm testing it

Reply:
No, this is necessary for UTs in the `core` module